scripts/screeps_rl_training_runner.py

```
@@ -781,48 +798,95 @@ def build_training_report(
+            if isinstance(raw_variants, list):
+                for raw_variant in raw_variants:
+                    if not isinstance(raw_variant, dict):
+                        continue
+                    variant_id = raw_variant.get("variant_id", raw_variant.get("variantId"))
+                    if not isinstance(variant_id, str):
+                        continue
+                    run_label = simulator_run_label(run, run_index)
+                    if variant_id not in expected_variant_id_set:
+                        raise RuntimeError(
+                            f"simulator run {run_label} emitted unexpected variant id {variant_id!r}"
+                        )
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)